client/src/pages/ManageSteps/DragDrop.js

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect } from 'react';
+import { DragDropContext, Draggable, Droppable } from 'react-beautiful-dnd';
+import { useHistory } from 'react-router-dom';
+import {
+  Grid as G,
+  Typography as T,
+  Icon,
+  Button,
+  Modal,
+} from '../../components';
+import { buttonStyle } from './style';
+import { navRoutes } from '../../constants';
+
+const columnsFromBackend = {
+  col1: {
+    name: 'Before claiming',
+    stage: 'beforeClaiming',
+    items: [],
+  },
+  col2: {
+    name: 'Claiming',
+    stage: 'claiming',
+    items: [],
+  },
+  col3: {
+    name: 'After claiming',
+    stage: 'afterClaiming',
+    items: [],
+  },
+};
+
+const reorderSteps = (cols) => {
+  let newSteps = [];
+  let cumulativeIndex = 0;
+  Object.keys(cols).forEach((key) => {
+    cols[key].items.forEach((item) => {
+      cumulativeIndex++;
+      newSteps.push({
+        ...item,
+        stage: cols[key].stage,
+        stepOrder: cumulativeIndex,
+      });
+    });
+  });
+  return newSteps;
+};
+
+const onDragEnd = (result, columns, setColumns) => {
+  if (!result.destination) return;
+  const { source, destination } = result;
+
+  if (source.droppableId !== destination.droppableId) {
+    const sourceColumn = columns[source.droppableId];
+    const destColumn = columns[destination.droppableId];
+    const sourceItems = [...sourceColumn.items];
+    const destItems = [...destColumn.items];
+    const [removed] = sourceItems.splice(source.index, 1);
+    destItems.splice(destination.index, 0, removed);
+    setColumns({
+      ...columns,
+      [source.droppableId]: {
+        ...sourceColumn,
+        items: sourceItems,
+      },
+      [destination.droppableId]: {
+        ...destColumn,
+        items: destItems,
+      },
+    });
+  } else {
+    const column = columns[source.droppableId];
+    const copiedItems = [...column.items];
+    const [removed] = copiedItems.splice(source.index, 1);
+    copiedItems.splice(destination.index, 0, removed);
+    setColumns({
+      ...columns,
+      [source.droppableId]: {
+        ...column,
+        items: copiedItems,
+      },
+    });
+  }
+};
+
+function DragDrop({ beforeClaiming, claiming, afterClaiming }) {
+  const [isModalVisible, setIsModalVisible] = useState(false);
```

Comment:
It feels slightly muddled here regarding the states here and in the parent state. If we think of this logically the parent state is going to be getting the steps from the server (I'm assuming this because you've currently got it getting the dummy data and then setting the original steps at this stage, then passing down as props). Therefore, on save changes, it would want to send to the server and then update the main state with the now updated steps from the server. 

So we need a save changes function in the parent component that would do this, and then on success show the modal or otherwise return an error message.

Reply:
done